src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ObligationsControllerTests.cs

```diff
@@ -212,5 +234,81 @@ public void PostSelectAuthority_WithInvalidModel_ReturnsHoldingViewWithViewModel
             UploadObligationsViewModel viewModel = viewResult.Model as UploadObligationsViewModel;
             Assert.NotNull(viewModel);
         }
+
+        [Fact]
+        public void UploadObligationsGet_IsDecoratedWith_HttpGetAttribute()
+        {
+            typeof(ObligationsController).GetMethod("UploadObligations", BindingFlags.Public | BindingFlags.Instance, null, CallingConventions.Any, new Type[] { typeof(CompetentAuthority) }, null)
+            .Should()
+            .BeDecoratedWith<HttpGetAttribute>();
+        }
+
+        [Fact]
+        public void UploadObligationsPost_IsDecoratedWith_HttpPostAttribute()
+        {
+            typeof(ObligationsController).GetMethod("UploadObligations", BindingFlags.Public | BindingFlags.Instance, null, CallingConventions.Any, new Type[] { typeof(UploadObligationsViewModel) }, null)
+            .Should()
+            .BeDecoratedWith<HttpPostAttribute>();
+        }
+
+        [Fact]
+        public void UploadObligationsPost_IsDecoratedWith_ValidateAntiForgeryTokenAttribute()
+        {
+            typeof(ObligationsController).GetMethod("UploadObligations", BindingFlags.Public | BindingFlags.Instance, null, CallingConventions.Any, new Type[] { typeof(UploadObligationsViewModel) }, null)
+            .Should()
+            .BeDecoratedWith<ValidateAntiForgeryTokenAttribute>();
+        }
+
+        [Fact]
+        public void DownloadTemplateGet_IsDecoratedWith_HttpGetAttribute()
+        {
+            typeof(ObligationsController).GetMethod("DownloadTemplate", BindingFlags.Public | BindingFlags.Instance, null, CallingConventions.Any, new Type[] { typeof(CompetentAuthority) }, null)
+            .Should()
+            .BeDecoratedWith<HttpGetAttribute>();
+        } 
+
+        [Fact]
+        public void UploadObligationsGet_PopulatesModel()
+        {
+            //arrange
+            var authority = CompetentAuthority.England;
+            var model = new UploadObligationsViewModel(authority);
+
+            //act
+            var result = controller.UploadObligations(authority) as ViewResult;
+
+            //assert
+            result.Model.Should().BeEquivalentTo(model);
+        }
+
+        [Fact]
+        public void UploadObligationsPost_SetsTriggerDownload_ToTrue()
+        {
+            //arrange
+            var authority = CompetentAuthority.England;
+            var model = new UploadObligationsViewModel(authority);
+
+            //act
+            var result = controller.UploadObligations(model) as ViewResult;
+
+            //assert
+            bool triggerDownload = result.ViewBag.TriggerDownload;
+
+            triggerDownload.Should().BeTrue();
+        }
+
+        [Fact]
+        public async Task DownloadTemplateGet_CallsGetPcsObligationsCsv_Once()
```

Comment:
would you mind updating it to ShouldCallHandler or something

Reply:
Updated